docs/source/core_classes_and_methods.rst

```diff
@@ -4,8 +4,9 @@ Core classes and methods
 
 .. currentmodule:: pints
 
-Pints provides the :class:`SingleOutputProblem` and
-:class:`MultiOutputProblem` classes to formulate
+Pints provides the :class:`SingleOutputProblem`,
+:class:`MultiOutputProblem`, :class:`ProblemCollection`
```

Comment:
Not a blocker: But can we think of some names that make it clear that:

1. A ProblemCollection can't just contain any old problem. We can't add singleoutput problems, for example
2. The ProblemCollection and SubProblem go together

I'm not sure what these names would be :D 
MultipartProblem and ProblemPart ?

Reply:
Maybe get the word "output" into the name for ProblemCollection, to show that all 3 problem types are actually classified by their output. So SingleOutput, MultiOutput, and... SplitOutputProblem? ComposedOutputProblem?

Reply:
Since the user never creates a SubProblem, (and since it is a very different thing from the other 3 classes called XProblem), is this a rare case for having a class inside a class?
E.g.
```
class ProblemCollection(object):
    ...
    
    class SubProblem(object):
        ...
        
    def subproblem(i):
        return ProblemCollection.SubProblem(...)
```